fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ComparisonPredicate.java

```diff
@@ -77,7 +77,7 @@ public boolean equals(Object o) {
             return false;
         }
         ComparisonPredicate other = (ComparisonPredicate) o;
-        return (type == other.getType()) && Objects.equals(left(), other.left())
+        return Objects.equals(left(), other.left())
```

Comment:
Could you explain why we do not compare type?

Reply:
Because it will be deleted